vra/data_source_fabric_compute.go

```diff
@@ -0,0 +1,162 @@
+	return &schema.Resource{
+		Read: dataSourceFabricComputeRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:          schema.TypeString,
+				ConflictsWith: []string{"filter"},
+				Description:   "The id of the fabric compute resource instance.",
+				Optional:      true,
+			},
+			"filter": {
+				Type:          schema.TypeString,
+				ConflictsWith: []string{"id"},
+				Description:   "Search criteria to narrow down the fabric compute resource instance.",
+				Optional:      true,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when the entity was created. The date is in ISO 8601 and UTC.",
+			},
+			"custom_properties": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "A list of key value pair of custom properties for the fabric compute resource.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A human-friendly description.",
+			},
+			"external_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id of the external entity on the provider side.",
+			},
+			"external_region_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The external region id of the fabric compute.",
+			},
+			"external_zone_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The external zone id of the fabric compute.",
+			},
+			"lifecycle_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Lifecycle status of the compute instance.",
+			},
+			"links": linksSchema(),
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A human-friendly name used as an identifier for the fabric compute resource instance.",
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id of the organization this entity belongs to.",
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Email of the user that owns the entity.",
+			},
+			"power_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Power state of fabric compute instance.",
+			},
+			"tags": tagsSchema(),
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of the fabric compute instance.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when the entity was last updated. The date is ISO 8601 and UTC.",
+			},
+		},
+	}
+}
+
+func dataSourceFabricComputeRead(d *schema.ResourceData, meta interface{}) error {
+	apiClient := meta.(*Client).apiClient
+
+	id := d.Get("id").(string)
+	filter := d.Get("filter").(string)
+
+	if id == "" && filter == "" {
+		return fmt.Errorf("one of id or filter is required")
```

Comment:
Since no argument is needed, could we use "errors" similar to https://github.com/vmware/terraform-provider-vra/pull/391/commits/38859a1e41d3dea671fe91231d82e413ad01383e

Reply:
yep, god catch!